security/pkg/nodeagent/sds/adapter.go

```diff
@@ -32,15 +32,15 @@ type sdsServiceAdapter struct {
 }
 
 func (d *sdsServiceAdapter) FetchSecrets(ctx context.Context, request *xdsapiv2.DiscoveryRequest) (*xdsapiv2.DiscoveryResponse, error) {
-	v3Resp, err := d.s.FetchSecrets(ctx, v2.UpgradeV2Request(request))
-	return v2.DowngradeV3Response(v3Resp), err
+	v3Resp, err := d.s.FetchSecrets(ctx, xds.UpgradeV2Request(request))
+	return xds.DowngradeV3Response(v3Resp), err
 }
 
 // We implement the v2 ADS API
 var _ discoveryv2.SecretDiscoveryServiceServer = &sdsServiceAdapter{}
 
 func (d *sdsServiceAdapter) StreamSecrets(stream discoveryv2.SecretDiscoveryService_StreamSecretsServer) error {
-	v3Stream := &v2.DiscoveryStreamV2Adapter{stream}
+	v3Stream := &xds.DiscoveryStreamV2Adapter{stream}
```

Comment:
Here the PR changes "v2. DiscoveryStreamV2Adapter" to "xds. DiscoveryStreamV2Adapter". DiscoveryStreamV2Adapter is specific to "istio.io/istio/pilot/pkg/proxy/envoy/v2" but "xds. DiscoveryStreamV2Adapter" does not contain the version. Can you explain how the version is represented in "xds. DiscoveryStreamV2Adapter"?

Reply:
The current code in v2 package is moved as is to xds. v2 package currently supports mostly v3 (which is the current envoy version that our code is mostly using). So `DiscoveryStreamV2Adapter` is adapter for v2 code.  In subsequent PR, I plan to move this to v2. To avoid too many changes, I moved every thing under v2 -> xds in this PR